scalanet/src/io/iohk/scalanet/peergroup/UDPPeerGroup.scala

```diff
@@ -187,14 +186,20 @@ class UDPPeerGroup[M](val config: Config)(
       val nettyChannel: NioDatagramChannel,
       localAddress: InetSocketAddress,
       remoteAddress: InetSocketAddress,
-      val messageSubject: ConnectableSubject[ChannelEvent[M]],
+      val messageQueue: ConcurrentQueue[Task, ChannelEvent[M]],
       channelType: UDPPeerGroupInternals.ChannelType
   ) extends Channel[InetMultiAddress, M] {
+    val eventLoopAsScheduler = monix.execution.Scheduler(nettyChannel.eventLoop())
 
     val closePromise: Promise[ChannelImpl] = nettyChannel.eventLoop().newPromise[ChannelImpl]()
 
     val channelId = UdpChannelId(nettyChannel.id(), remoteAddress, localAddress)
 
+    def pushEventOnNettyScheduler(ev: ChannelEvent[M]): Unit = {
+      println("Push messages on")
+      messageQueue.offer(ev).runSyncStep(eventLoopAsScheduler)
+    }
```

Comment:
Remove `println`? I think it's unnecessary to state that pushing happens on the netty scheduler unless the caller has a choice of another method to call. 

To be clear, the processing of the events will not happen on the netty event loop, right? 

We have to make sure that this step doesn't block because it would hold up the reading/writing of data across all Netty channels.

Reply:
sure, println was only for some debugging purposes only.

Yep processing of events will happen during observable execution on user scheduler. So this method will not block, if the queue is unboudned.

Pushing event on netty thread, makes is possible not to provide external scheduler to peergroup, also it should be more performant as there is no context switch i.e the same thread which received message on netty side pushes it to queue.